db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -11599,6 +11665,115 @@ FromTable tableFactor() throws StandardException :
                                 getContextManager());
         return fromTable;
     }
+|
+    fromTableType = fromTableType() <TABLE> <LEFT_PAREN> statement = preparableDMLDataStatement() <RIGHT_PAREN>
+    [ optionalTableClauses = optionalTableProperties() ]  {
+        boolean isOld = fromTableType.kind == OLD;
+        boolean isFinalTable = fromTableType.kind == FINAL;
+        dmlStatement = (DMLModStatementNode)statement;
+        LanguageConnectionContext lcc = dmlStatement.getLanguageConnectionContext();
+        tableName = dmlStatement.getTargetTableName();
+        String uniqueId = java.util.UUID.randomUUID().toString().replaceAll("-", "_").toUpperCase();
+        String tempTriggerName = "TRIG_" + uniqueId;
```

Comment:
It doesn't matter what name we use here since the trigger and referenced table (NT_<>) are only visible to the executing query.  I could have just called these TRIGGER and NT, but having a really long name makes it clear that this is the temporary trigger, for debugging purposes.  I guess I could have used INTERNALLY_GENERATED_TRIGGER and FROM_TABLE_DML_REFERENCED_TABLE as names to make it clearer.